code/core/src/phone/java/com/adobe/marketing/mobile/internal/util/ContextExt.kt

```diff
@@ -0,0 +1,28 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.internal.util
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+
+/**
+ * An extension for finding the activity from a context.
+ */
+internal fun Context.findActivity(): Activity? {
+    var context = this
+    while (context is ContextWrapper) {
+        if (context is Activity) return context
+        context = context.baseContext
+    }
+    return null
+}
```

Comment:
Since this is only used in the UI area, can we make this a file private function inside `MessageFrame.kt` for now? We can move this to broader utils later when we find its uses at other places.

Reply:
sounds good, will move it